python/nano/notebooks/nano-trainer-example.ipynb

```diff
@@ -0,0 +1,329 @@
+    "                steps_per_epoch=steps_per_epoch,\n",
+    "            ),\n",
+    "            \"interval\": \"step\",\n",
+    "        }\n",
+    "        return {\"optimizer\": optimizer, \"lr_scheduler\": scheduler_dict}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "seed_everything(7)\n",
+    "PATH_DATASETS = os.environ.get(\"PATH_DATASETS\", \".\")\n",
+    "BATCH_SIZE = 64\n",
+    "NUM_WORKERS = int(os.cpu_count() / 2)\n",
+    "data_module = prepare_data(PATH_DATASETS, BATCH_SIZE, NUM_WORKERS)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train\n",
+    "Use Trainer from bigdl.nano.pytorch.trainer for BigDL-Nano pytorch.\n",
+    "\n",
+    "This Trainer extends PyTorch Lightning Trainer by adding various options to accelerate pytorch training.\n",
+    "\n",
+    "```\n",
+    "    :param num_processes: number of processes in distributed training. default: 4.\n",
+    "    :param use_ipex: whether we use ipex as accelerator for trainer. default: True.\n",
+    "    :param cpu_for_each_process: A list of length `num_processes`, each containing a list of\n",
+    "            indices of cpus each process will be using. default: None, and the cpu will be\n",
+    "            automatically and evenly distributed among processes.\n",
+    "```\n",
+    "The next few cells show examples of different parameters.\n",
+    "#### Single Process\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = LitResnet(learning_rate=0.05)\n",
+    "model.datamodule = data_module\n",
+    "basic_trainer = Trainer(num_processes = 1,\n",
+    "                  use_ipex = False,\n",
+    "                  progress_bar_refresh_rate=10,\n",
+    "                  max_epochs=30,\n",
+    "                  logger=TensorBoardLogger(\"lightning_logs/\", name=\"basic\"),\n",
+    "                  callbacks=[LearningRateMonitor(logging_interval=\"step\")])\n",
+    "start = time()\n",
+    "basic_trainer.fit(model, datamodule=data_module)\n",
+    "basic_fit_time = time() - start\n",
+    "outputs = basic_trainer.test(model, datamodule=data_module)\n",
+    "basic_acc = outputs[0]['test_acc'] * 100"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### multi processes\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = LitResnet(learning_rate=0.05)\n",
+    "model.datamodule = data_module\n",
+    "mulit_trainer = Trainer(num_processes=2,\n",
+    "                        use_ipex = False,\n",
+    "                        distributed_backend=\"subprocess\",\n",
+    "                        progress_bar_refresh_rate=10,\n",
+    "                        max_epochs=3,\n",
```

Comment:
why this trainer only has 3 epochs and other trainers have 30 epochs.

Reply:
That's a mistake. I set max_epochs =3 when I debug the code. It should be 30 as same as other trainers.